great_expectations/datasource/fluent/data_asset/data_connector/file_path_data_connector.py

```diff
@@ -73,6 +73,7 @@ def __init__(
         # sorters: Optional[list] = None,
         # TODO: <Alex>ALEX</Alex>
         file_path_template_map_fn: Optional[Callable] = None,
+        get_unfiltered_batch_definition_list_fn: None | Callable = None,
```

Comment:
@billdirks, just a question for my own understanding - when adding types to the Callable annotation it looks like [you can only use positional arguments](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#callable-types-and-lambdas) with the function otherwise mypy will complain that keyword arguments are supplied. Should we be moving toward positional arguments instead of keyword? Also `directory_get_unfiltered_batch_definition_list_fn` takes an additional parameter `data_directory: str` - is there a way to handle that in the type hint?

Reply:
Oh, I didn't see this comment before. I think we've already answered your second one. For your first question (positional vs keyword arguments), I think how we are doing it now is fine. This has to do with how the function is defined vs called. Calling the function with keyword arguments is clear. `Callable` enforces the order of the inputs/outputs must be fixed vs specifying (name, type) pairs. That doesn't seem like a burden for us so is fine too. Hopefully, this answers your question. We can chat synchronously if that would help.

Reply:
This is great - thank you! I agree that what we now have in the PR works. We can revisit if there are methods that we want to call with keyword arguments.